raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -3788,23 +3787,7 @@ public void complete() {
         }
     }
 
-    private static final class Registration<T> {
-        private final Ops ops;
-        private final Listener<T> listener;
-
-        private Registration(Ops ops, Listener<T> listener) {
-            this.ops = ops;
-            this.listener = listener;
-        }
-
-        private Ops ops() {
-            return ops;
-        }
-
-        private Listener<T> listener() {
-            return listener;
-        }
-
+    private record Registration<T>(KafkaRaftClient.Registration.Ops ops, Listener<T> listener) {
```

Comment:
The `record` keyword starts support in jdk 14, but client is still supporting jdk 11 (raft module will be used by `tool` module). So I think it'd be better we revert this change for now.

Reply:
Right, reverted. Thanks.